modules/aruco/src/aruco.cpp

```diff
@@ -1542,22 +1542,22 @@ void _drawPlanarBoardImpl(Board *_board, Size outSize, OutputArray _img, int mar
             // remove negativity
             p1.x = p0.x - minX;
             p1.y = p0.y - minY;
-            pf.x = p1.x * float(markerZone.cols - 1) / sizeX;
-            pf.y = float(markerZone.rows - 1) - p1.y * float(markerZone.rows - 1) / sizeY;
+            pf.x = p1.x * float(markerZone.cols) / sizeX;
```

Comment:
Why?
- `p0` is the coordinate in the Board ref system
- `p1` is the coordinate in the Board ref system after removing negativity
- `pf` is the coordinate in the image ref system (in pixels)
- `sizeX` is the total range in Board coordinates (for x axis)
- `markerZone.cols` is the total range in image coordinates (for x axis in pixels)

In this line we are just converting from `p1` to `pf`. This is scaling with the rate `markerZone.cols/sizeX`